### ISSUE TITLE
Issue With OpenShift : rootless ?

### ISSUE THREAD

**Issue by gmoirod** [bitnami/bitnami-docker-discourse#171]
**Description**

When I tried to install the discourse chart, which I think uses the image here, in openshift OCP 4.4 I got an error that reads:

```
Starting discourse-sidekiq...
ERROR Cannot find any module matching the provided specification 'discourse-sidekiq'
```

```
Starting discourse...
ERROR Cannot find any module matching the provided specification 'discourse'
```

I suppose this is like https://github.com/bitnami/bitnami-docker-wordpress/issues/127 where containers cannot run rootless.

**Steps to reproduce the issue:**

    Setup openshift 3.9 (I used minishift)
    Use helm install stable/wordpress to start it up
    Read the logs to the wordpress pod that didn't come up properly

1. Setup openshift OCP 4.4
2. Use helm install discourse bitnami/discourse
3. Read the logs to the discourse pod that didn't come up properly

**Comment by javsalgar** [comment]
Hi,

Thanks for reporting. Right now our Discourse container is a `root` container. We're working on moving our web application containers to non-root (just like our infrastructure ones). Currently there are no short-term plans to move this one to non-root but it is in our backlog. We will update this ticket when the container is moved to non-root.

**Comment by javsalgar** [comment]
Not stale

**Comment by sbkg0002** [comment]
Any progress on this? We would also like this.

**Comment by javsalgar** [comment]
Hi,

I'm afraid that it is still in our backlog. Currently, our efforts are focused on moving the PHP application to non-root (and also to containers fully based on bash). I imagine that, once we finish them, we will go to Ruby applications.

**Comment by RaphaelMui** [comment]
Hi @javsalgar,

May I know the necessary steps to make the Discourse container non-root?

**Comment by javsalgar** [comment]
Hi,

It is not trivial. The first thing would be to change the user to 1001, and adapt the permissions of the folders so the container works as non-root. You could try changing the user and see the effects in the container. However, I believe it will require extra work from us in the logic.

**Comment by RaphaelMui** [comment]
Hi @Javslgar,

Yes, I need to create the discourse user and group and 'chown' manually.  It seems to initialize discourse successfully, but it cannot be started.  The error is thrown from nami.  How can I configure nami to not start discourse as root?

```
discour INFO  Preparing Discourse environment...
discour INFO  Configuring the Discourse database files...
discour INFO  Configuring Redis...
discour INFO  Setting up SMTP in Discourse...
discour INFO  Migrating database...
discour INFO 
discour INFO  ########################################################################
discour INFO   Installation parameters for discourse:
discour INFO     Persisted data and properties have been restored.
discour INFO     Any input specified will not take effect.
discour INFO   This installation requires no credentials.
discour INFO  ########################################################################
discour INFO 
nami    INFO  discourse successfully initialized
Starting discourse...
nami    ERROR You have to be root to call start`
```

**Comment by javsalgar** [comment]
I'm afraid it would require to change the node.js logic in `/root/.nami/components`. During these two weeks we will change the container to be bash and non-root. My suggestion would be to wait for the new container instead.

**Comment by RaphaelMui** [comment]
This is really good news.   thx @javsalgar

**Comment by marcosbc** [comment]
Hi all, unfortunately non-root is not going to be included in the planned changes because of Passenger limitations.

Note that the logic will be prepared to initialize as root. However, the problem is starting Discourse with Passenger, which is the tool we are using to run the application. Refer to [this site](https://www.phusionpassenger.com/documentation/Security%20of%20user%20switching%20support.html) for more context.

We expect the Discourse release with open-source logic in the [`rootfs` folder](https://github.com/bitnami/bitnami-docker-discourse/tree/master/2/debian-10/rootfs) of this repo to be published throughout the following week. We will let you know when that happens.

Once it is out, you could try and modify the `run.sh` script used to start Discourse with a different application server, like Puma or even `rails server`. That way you may be able to get it working as a non-root user.

**Comment by marcosbc** [comment]
Hi, FYI we just released a new container version, [2.7.0-debian-10-r4](https://github.com/bitnami/bitnami-docker-discourse#270-debian-10-r4). The initialization logic has received a major refactor, and its scripts are now located in the [`rootfs/` directory](https://github.com/bitnami/bitnami-docker-discourse/blob/master/2/debian-10/rootfs/).

Unfortunately we have not made progress for this specific issue, but now it will be a lot easier if you want to contribute non-root support. Note that you may need to change the server to something like Puma, as I mentioned earlier. Any contributions in that direction are welcome and we'll be glad to help with the release.

**Comment by carrodher** [comment]
Hi, unfortunately, the task is still on our backlog and there are no plans to work on it in the short/middle term.

In the same way, please, [note the Bitnami team is not actively working on new features/improvements nor providing support through GitHub Issues for this project](https://github.com/bitnami/bitnami-docker-discourse#community-supported-solution) since this asset was marked as a “Community supported solution”.

At this point, the best option to fix any issue or add a new feature is to contribute create a PR; the Bitnami team will be happy to review it and provide feedback.
